### PR TITLE
Snapshot online vm

### PR DESCRIPTION
**Pull request by ShellyKa13**
**What this PR does / why we need it**:
Currently online snapshot of VM was prevented. This PR removed the restrictions and allows it.
Added and changed tests to check and support online vm snapshot.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
Tested on multiple host and guest file systems. It does not support xfs as host FS.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-controller/watch/snapshot/snapshot.go [hunk 1]
```diff
@@ -872,7 +872,6 @@ func (s *vmSnapshotSource) Lock() (bool, error) {
 
 	if vmRunning {
 		log.Log.V(3).Infof("Snapshotting a running VM is not supported yet")
```
Comment (mhenriks) [inline pkg/virt-controller/watch/snapshot/snapshot.go:874]: Delete/change this log line

File: pkg/virt-controller/watch/snapshot/snapshot_test.go [hunk 2]
```diff
@@ -674,6 +710,37 @@ var _ = Describe("Snapshot controlleer", func() {
 				controller.processVMSnapshotWorkItem()
 			})
 
+			It("should update VirtualMachineSnapshotStatus Indications if vm is running", func() {
```
Comment (mhenriks) [inline pkg/virt-controller/watch/snapshot/snapshot_test.go:713]: I can't find where indications are added to the snapshot

**Review by mhenriks — COMMENTED** [review]
So far so good 👍

File: pkg/virt-controller/watch/snapshot/snapshot_test.go (hunk 2 above)
Comment (ShellyKa13) [reply pkg/virt-controller/watch/snapshot/snapshot_test.go:713]: it should be added in the code itself and then be compared to the "updatedSnapshot" created in line 721, which we update its expected indications in lines 727-728

File: pkg/virt-controller/watch/snapshot/snapshot.go (hunk 1 above)
Comment (ShellyKa13) [reply pkg/virt-controller/watch/snapshot/snapshot.go:874]: right, will do

**Comment by mhenriks** [comment]
👍 👍

**Pull request merged**